src/main/java/com/uber/cadence/worker/Worker.java

```diff
@@ -344,4 +353,86 @@ public String toString() {
   public String getTaskList() {
     return taskList;
   }
+
+  public static final class Factory {
+
+    private final ArrayList<Worker> workers = new ArrayList<>();
+    private final Supplier<IWorkflowService> getWorkFlowService;
+    private String domain;
+    private State state = State.Initial;
+
+    private final String statusErrorMessage =
+        "attempted to %s while in %s state. Acceptable States: %s";
+
+    public Factory(String domain) {
+      this(() -> new WorkflowServiceTChannel(), domain);
+    }
+
+    public Factory(String host, int port, String domain) {
+      this(() -> new WorkflowServiceTChannel(host, port), domain);
+    }
+
+    public Factory(Supplier<IWorkflowService> getWorkFlowService, String domain) {
```

Comment:
What's the use case for passing Supplier instead of the IWorkflowService instance?

Reply:
Nope, changed this to passing in an instance. The factory will not be responsible for closing the instance passed in.